### DISCUSSION TITLE
Windows Virtual Desktop - Problem z rozszerzeniem joindomain

### DISCUSSION THREAD

**Discussion by arkadiuszstrzezek** [AzureCommunityPL/group#145] [Pomoc]
Cześć,

Mam pewien problem związany z Azure WVD. Dokładniej sprawa wygląda następująco. Po przechwyceniu obrazu z maszyny i próbie uruchomienie Deploymentu WVD, w trakcie instalacji rozszerzenia joindomain dostaje błąd 

"message": "VM-0' has not reported status for VM agent or extensions. Verify that the OS is up and healthy, the VM has a running VM agent, and that it can establish outbound connections to Azure storage. Please refer to https://aka.ms/vmextensionwindowstroubleshoot for additional VM agent troubleshooting information."

Czy spotkaliście się może z podobnym problemem? dodam, że po uruchomieniu samej maszyny, z tego obrazu, jej agent zwraca status not ready.
Pomijam fakt sieci itd, ponieważ komunikacja z kontrolerem DC została przetestowana i działa tak samo konta. z góry dziękuję za pomoc :)

**Comment by przybylskirobert** [comment]
Hej ponizej kilka pytań ktore pomogą w rozwiązaniu (o ile jeszcze nie uporales sie z tym)
1. Której wersji OS dotyczy problem ? ( Windows 10 / Server, osbuild 2004 ?, multisession czy zwykły OS)
2. Image z marketplace czy customowy?
3. Jesli customowy Image to czy zrobiles sysprep i czy przed sysprep VM byla poza domeną?
4. Czy deployment robisz ręcznie czy przez devops / skryptem ?

**Comment by arkadiuszstrzezek** [comment]
Hej,
Niestety problem dalej występuje, poniżej załączam odpowiedzi na pytania:
1) Problem dotyczy wersji 2004. Windows 10 Enterprise for Virtual Desktop
2) Obraz: pochodzi z Azure marketplace, wdrożone fslogix, wdrożona aplikacja kliencka
3) VM stała w domenie przed sysprep
4) Deployment wykonuje ręcznie z potralu WVD

**Reply by przybylskirobert** [reply]
Czy probowales moze sprawdzic najprostrze rozwiazanie tj redeploy session hosta / ewentualnie deploy session hosta z obrazem z marketplace bez zadnych dodatkow - w celu sprawdzenia czy doda sie poprawnie do domeny. Zakladam ze masz Active Directory a nie Azure AD DS ? 
W przeszlosci mialem podobny case i redeploy rozwiazal problem.

**Reply by arkadiuszstrzezek** [reply]
W moim scenariuszu korzystam z Azure AD DS. Niestety redeploy tez nie pomaga w rozwiązaniu problemu.

Najbardziej zaskakujące jest to, że po stworzeniu pojedynczej maszyny wirtualnej z przygotowanego obrazu agent VM zgłasza status - Not Ready.  Po zalogowaniu się do maszyny okazało się, że usługa WindowsAzureGuestAgent jest zatrzymana i dopiero po jej restarcie maszyna nawiązuje komunikacje i można wykonać na niej chociażby connection test.

**Reply by przybylskirobert** [reply]
Skoro uzywasz Azure AD DS, czy po deploymencie Azure AD DS zmieniles haslo usera ktorym chcesz dodac komputery do domeny ? 
Po synchronizacji userów z AAd do Azure AD DS wymagana jest zmiana hasla.
Raz o tym zapomnialem i mialem podobny błąd, wiecej info tutaj: https://docs.microsoft.com/en-us/azure/active-directory-domain-services/tutorial-create-instance#enable-user-accounts-for-azure-ad-ds

**Reply by przybylskirobert** [reply]
czy udalo sie rozwiazac problem ?

**Comment by rabusiak** [comment]
Hejka, ja też miałem różne przygody ostatnio z wvd. Polecam sobie z boku przygotować czyste vmki wpięte w domenę. Następnie ręcznie dodać je do puli kluczem rejestracyjnym. Aplikacje i sam system to już sobie ansible, puppetem czy chociażby dsc dopracujesz.
Jest z tą technologią spory bałagan, bo większość instrukcji dotyczy starszej wersji - obecnie nie trzeba tworzyć żadnych tenantów, ja sam pomieszałem wszystko i po dniu walki zaorałem i zacząłem od nowa xD

PS. FsLogicAps udało Ci się zrobić z dołączeniem storage accounta do AD? Ja po kilku godzinach się poddałem, niby dodawał się do domeny, obiekt w ad się pojawiał, ale wywalało na koniec - brak uprawnień (kontem enterprise/domain admina).
Skończyło się na udziale w onpremowym fs ;)

**Reply by przybylskirobert** [reply]
Odnosnie FSLogix - czy po dodaniu do domeny storage account:
nadales uprawnienia NTFS z poziomu share oraz z poziomu Azure
nadales dostep rolom Storage File DATA SMB Share Contributor oraz Storage File DATA SMB Share Elevated Contributor
konfiguracje wpisow w rejestrze dla FSLogix tez zrobiles poprawnie ?

**Reply by rabusiak** [reply]
Problem pojawiał się już na etapie dodawania do domeny (obiekt się tworzył, ale polecenie wywalało błąd).
https://github.com/Azure-Samples/azure-files-samples/issues/54
